// @portabletext/editor/lib/_chunks-cjs/util.split-text-block.cjs
"use strict";
var util_selectionPointToBlockOffset = require("./util.selection-point-to-block-offset.cjs"), util_sliceBlocks = require("./util.slice-blocks.cjs");
function isTextBlock(context, block) {
  return block._type === context.schema.block.name;
}
function mergeTextBlocks({
  context,
  targetBlock,
  incomingBlock
}) {
  const parsedIncomingBlock = util_selectionPointToBlockOffset.parseBlock({
    context,
    block: incomingBlock,
    options: {
      refreshKeys: !0
    }
  });
  return !parsedIncomingBlock || !isTextBlock(context, parsedIncomingBlock) ? targetBlock : {
    ...targetBlock,
    children: [...targetBlock.children, ...parsedIncomingBlock.children],
    markDefs: [...targetBlock.markDefs ?? [], ...parsedIncomingBlock.markDefs ?? []]
  };
}
function splitTextBlock({
  context,
  block,
  point
}) {
  const firstChild = block.children.at(0), lastChild = block.children.at(block.children.length - 1);
  if (!firstChild || !lastChild)
    return;
  const before = util_sliceBlocks.sliceBlocks({
    blocks: [block],
    selection: {
      anchor: {
        path: [{
          _key: block._key
        }, "children", {
          _key: firstChild._key
        }],
        offset: 0
      },
      focus: point
    }
  }).at(0), after = util_sliceBlocks.sliceBlocks({
    blocks: [block],
    selection: {
      anchor: point,
      focus: {
        path: [{
          _key: block._key
        }, "children", {
          _key: lastChild._key
        }],
        offset: util_sliceBlocks.isSpan(context, lastChild) ? lastChild.text.length : 0
      }
    }
  }).at(0);
  if (!(!before || !after) && !(!isTextBlock(context, before) || !isTextBlock(context, after)))
    return {
      before,
      after
    };
}
exports.isTextBlock = isTextBlock;
exports.mergeTextBlocks = mergeTextBlocks;
exports.splitTextBlock = splitTextBlock;
